plugins/main/public/components/common/data-source/pattern/pattern-data-source-filter-manager.ts

```diff
@@ -394,24 +395,29 @@ export class PatternDataSourceFilterManager
   /********************** FILTERS FACTORY ***************************/
   /******************************************************************/
 
-  /**
-   * Returns a filter with the field and value received
-   * @param field
-   * @param value
-   * @returns
-   */
-  createFilter(
+  static createFilter(
     type: tFilterType,
     key: string,
     value: string | [],
+    indexPatternId: string,
     controlledBy?: string,
-  ): tFilter {
-    switch (type) {
+  ) {
+    if (!key || !value || !indexPatternId) {
+      throw new Error('key, value and indexPatternId params are required');
+    }
+
+    if (type === 'is one of' || type === 'is not one of') {
```

Comment:
You are using strings as operators and it's used several times. I think you could improve this by defining an enum and exporting it, so the developers don't really need to know how the operator string is. E.g

```
enum FILTER_OPERATOR {
  IS = 'is',
  IS_NOT = 'is not',
  EXISTS = 'exists',
  DOES_NOT_EXISTS = 'does not exists',
  IS_ONE_OF = 'is one of',
  IS_NOT_ONE_OF = 'is not one of',
}
```

So if the string of the operator changes in the future we won't have to change all the implementations, except for the enum.

Reply:
Changed in https://github.com/wazuh/wazuh-dashboard-plugins/pull/6745/commits/bbfb7a2a794acf22a1682d447a2c0494a6b836e4